File: summary/rust/7_greatest_common_divisor.rs
fn gcd(a: u64, b: u64) -> u64 {
    if b == 0 {
        return a;
    } else {
        return gcd(b, a % b);
    }
}

fn main() {
    let a = 24;
    let b = 18;
    let result = gcd(a, b);
    println!("{} と {} の最大公約数は {}", a, b, result);
}

File: summary/python/7_greatest_common_divisor.py
def gcd(a, b):
    while a >= 1 and b >= 1:
        if a >= b:
            a = a % b
        else:
            b = b % a
    return a if a >= 1 else b

# 入力を受け取る
a, b = map(int, input().split())

# 最大公約数を計算して出力
result = gcd(a, b)
print(f"{result}")
